File: app/src/main/java/com/akrivonos/a2chparser/utils/ItemDecoratorUtils.java
package com.akrivonos.a2chparser.utils;

import android.graphics.Rect;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

public class ItemDecoratorUtils {

    public static RecyclerView.ItemDecoration createItemDecorationOffsets(final DecorationDirection direction, final int countDecoration) {
        return new RecyclerView.ItemDecoration() {
            @Override
            public void getItemOffsets(@NonNull Rect outRect, @NonNull View view, @NonNull RecyclerView parent, @NonNull RecyclerView.State state) {
                super.getItemOffsets(outRect, view, parent, state);
                switch (direction) {
                    case TOP:
                        outRect.top = countDecoration;
                        break;
                    case LEFT:
                        outRect.left = countDecoration;
                        break;
                    case RIGHT:
                        outRect.right = countDecoration;
                        break;
                    case BOTTOM:
                        outRect.bottom = countDecoration;
                        break;
                }
            }
        };
    }

    public enum DecorationDirection {
        TOP, BOTTOM, RIGHT, LEFT
    }
}

File: app/src/main/java/com/akrivonos/a2chparser/utils/ItemDecoratorUtils.kt
package com.akrivonos.a2chparser.utils

import android.graphics.Rect
import android.view.View
import androidx.recyclerview.widget.RecyclerView
import javax.inject.Inject

class ItemDecoratorUtils @Inject constructor() {

    fun createItemDecorationOffsets(direction: DecorationDirection, countDecoration: Int): RecyclerView.ItemDecoration {
        return object : RecyclerView.ItemDecoration() {
            override fun getItemOffsets(outRect: Rect, view: View, parent: RecyclerView, state: RecyclerView.State) {
                super.getItemOffsets(outRect, view, parent, state)
                when (direction) {
                    DecorationDirection.TOP -> outRect.top = countDecoration
                    DecorationDirection.LEFT -> outRect.left = countDecoration
                    DecorationDirection.RIGHT -> outRect.right = countDecoration
                    DecorationDirection.BOTTOM -> outRect.bottom = countDecoration
                }
            }
        }
    }

    enum class DecorationDirection {
        TOP, BOTTOM, RIGHT, LEFT
    }
}
